src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -108,10 +108,18 @@ private void processZipFile(CloudBlobContainer container, String zipFilename)
 
             Envelope envelope = envelopeProcessor.processEnvelope(metadataStream);
 
-            documentProcessor.processPdfFiles(pdfFiles, envelope.getScannableItems());
-            envelopeProcessor.markAsUploaded(envelope, container.getName(), zipFilename);
-
-            cloudBlockBlob.delete();
+            try {
+                documentProcessor.processPdfFiles(pdfFiles, envelope.getScannableItems());
+                envelopeProcessor.markAsUploaded(envelope, container.getName(), zipFilename);
+
+                cloudBlockBlob.delete();
+            } catch (Exception exception) {
+                // catching any exception originated from document processor
```

Comment:
It would be nice if we could log envelope name along with exception just in case update to DB fails ?

Reply:
logging should come in together with exception handler up in another level. this additional "exception wrapping" is for recording states. left logging out of it intentionally